### DISCUSSION TITLE
How to use default filter on table

### DISCUSSION THREAD

**Discussion by webparadoxer** [filamentphp/filament#551] [Help] [answered]
Hi!
How I can filter resource model by query?
Like this - select rows by auth user_id

**Comment by danharrin** [comment, marked as answer]
Yes, you can override the `getQuery()` static method on the List page to scope the Eloquent query.

**Reply by underdpt** [reply]
Hello,

I'm trying to use a default filter, and this solution leads to another issues. This is my use case: i have a list of products, some updated and others to be updated. I want to show by default only the products to be updated, and add one filter to show the products that have been updated.

With this solution, the "default filter" is always enabled, almost like a global scope and without a way to disable it. Being `getQuery()` a static function, I cannot tinker on the `ListRecords` class to modify the default query based on the `$filter` property to check if there's another filter in place to return the original query.

I think the best would be to add a `->default()` method to the filters, and a way to make that default filter to execute if no other filter are in place. I think I can add this easily by modifying `applyFilters()` on `vendor\filament\filament\packages\tables\src\Concerns\CanFilterRecords.php`, so if you think this solution fits i can make a PR.

**Reply by danharrin** [reply]
Hey @underdpt, sure, go for it :)

**Reply by underdpt** [reply]
Done, hope it helps!

https://github.com/laravel-filament/filament/pull/736
